src/components/QuickOpenModal.js

```diff
@@ -0,0 +1,378 @@
+      query,
+      searchType
+    } = this.props;
+    const { results, selectedIndex } = this.state;
+
+    if (!enabled || !results) {
+      return;
+    }
+
+    if (e.key === "ArrowUp") {
+      this.traverseResults(-1);
+    } else if (e.key === "ArrowDown") {
+      this.traverseResults(1);
+    } else if (e.key === "Enter") {
+      switch (searchType) {
+        case "goto":
+          if (!selectedSource) {
+            return;
+          }
+          const [, line, column] = query.split(":");
+          const lineNumber = parseInt(line, 10);
+          const columnNumber = parseInt(column, 10);
+          if (!isNaN(lineNumber)) {
+            selectSource(selectedSource.get("id"), {
+              location: {
+                line: lineNumber,
+                ...(!isNaN(columnNumber) ? { column: columnNumber } : null)
+              }
+            });
+            this.closeModal();
+            return;
+          }
+          this.closeModal();
+          return;
+        case "sources":
+        case "functions":
+        case "variables":
+        default:
+          this.selectResultItem(e, results[selectedIndex]);
+          this.closeModal();
+          return;
+      }
+    } else if (e.key === "Tab") {
+      this.closeModal();
+    }
+  };
+
+  resultCount() {
+    return this.state.results ? this.state.results.length : 0;
+  }
+
+  renderResults() {
+    const { enabled, searchType } = this.props;
+    const { selectedIndex, results } = this.state;
+
+    if (!enabled || !results) {
+      return null;
+    }
+
+    return (
+      <ResultList
+        key="results"
+        items={results}
+        selected={selectedIndex}
+        selectItem={this.selectResultItem}
+        ref="resultList"
+        {...(searchType === "sources" || searchType === "gotoSource"
+          ? { size: "big" }
+          : {})}
+      />
+    );
+  }
+
+  renderInput() {
+    const { query, searchType } = this.props;
+    const summaryMsg = L10N.getFormatStr(
+      "sourceSearch.resultsSummary1",
+      this.resultCount()
+    );
+
+    const showSummary =
+      searchType === "sources" ||
+      searchType === "functions" ||
+      searchType === "variables";
+
+    return (
+      <div key="input" className="input-wrapper">
+        <SearchInput
+          query={query}
+          count={this.resultCount()}
+          placeholder={L10N.getStr("sourceSearch.search")}
+          {...(showSummary === true ? { summaryMsg } : {})}
+          onChange={this.onChange}
+          onKeyDown={this.onKeyDown}
+          handleClose={this.closeModal}
+        />
+      </div>
+    );
+  }
+  render() {
+    const { enabled } = this.props;
+
+    if (!enabled) {
+      return null;
+    }
+
+    return (
+      <Modal in={enabled} handleClose={this.closeModal}>
+        {this.renderInput()}
+        {this.renderResults()}
+      </Modal>
+    );
+  }
+}
+
+export default connect(
+  state => {
+    const source = getSelectedSource(state);
```

Comment:
minor thing, but lets move this to a `mapStateToProps` function

Reply:
👍